tests/tcp.rs

```diff
@@ -67,20 +73,30 @@ pub fn should_work_with_multiple_fds() {
     s2.set_nonblocking(true).expect("set_nonblocking call failed");
     s3.set_nonblocking(true).expect("set_nonblocking call failed");
 
+    // Wait for the server to send the data
     client_notification.recv().expect("receive server notifcation");
-    selector.add_read(&s1);
-    selector.add_read(&s2);
-    selector.add_read(&s3);
-    let result = selector.select().expect("To try select");
-    assert_eq!(result.len(), 2);
-    assert!(!result.is_read(&s1));
-    assert!(result.is_read(&s2));
-    assert!(result.is_read(&s3));
 
+    loop {
+        selector.add_read(&s1);
```

Comment:
`add_read` needs to happen once
`selector` is not modified by `select()` call and instead returns modified copy